ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -254,18 +250,35 @@ void setFlag(Flags flag) {
     flags = (byte) (flags | (1 << flag.ordinal()));
   }
 
+  /**
+   * Clears the provided {@code flag}
+   * @param flag the flag to clear
+   */
+  void clearFlag(Flags flag) {
```

Comment:
Hesitant to make the change in this PR since it will change about 141 lines. Can make the change as a final commit before the merge if that is ok with you